src/redfish_collector/core/dataReconstruction.py

```
@@ -15,7 +15,11 @@ def remove_odata_elements(d,notUseValue):
 
 def fixListConverter(data):
     if isinstance(data, dict):
-        if all(isinstance(member, int) for member in data.keys()):
+        # NOTE: `data and ...` is required — all() of an empty dict's keys is True,
+        # which would silently turn an empty {} into [], flipping the whole
+        # reconstructed structure from dict to list and crashing downstream
+        # (collectedData['Common'] -> TypeError). Empty dict must stay a dict.
+        if data and all(isinstance(member, int) for member in data.keys()):
```

Ruff (0.15.17)

[warning] Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)